inspirehep/modules/literaturesuggest/tasks.py

```diff
@@ -24,205 +24,220 @@
+        page_start=page_start,
+        page_end=page_end,
+        artid=artid
+    )
+
+    if form_fields.get('preprint_created'):
+        builder.add_imprint_date(form_fields.get('preprint_created'))
+
+    builder.add_thesis(
+        defense_date=form_fields.get('defense_date'),
+        degree_type=form_fields.get('degree_type'),
+        institution=form_fields.get('institution'),
+        date=form_fields.get('thesis_date')
+    )
+    if form_fields.get('experiment'):
+        builder.add_accelerator_experiment(experiment=form_fields.get('experiment'))
+
+    language = _language_mapping(
+        form_fields.get('language'),
+        form_fields.get('other_language')
+    )
+    if language and len(language) == 2:
+        builder.add_language(language=language)
+
+    if form_fields.get('title_translation'):
+        builder.add_title_translation(
+            title=form_fields.get('title_translation'),
+            language=language
+        )
+
+    if form_fields.get('title_arXiv'):
+        builder.add_title(
+            title=form_fields.get('title_arXiv'),
+            source='arXiv'
+        )
+
+    if form_fields.get('title_crossref'):
+        builder.add_title(
+            title=form_fields.get('title_crossref'),
+            source='crossref'
+        )
+
+    if form_fields.get('license_url'):
+        builder.add_license_url(license_url=form_fields.get('license_url'))
+
+    if form_fields.get('public_notes'):
+        builder.add_public_note(form_fields.get('public_notes'))
+
+    if form_fields.get('note'):
+        builder.add_public_note(
+            form_fields.get('note'),
+            source='arXiv' if form_fields.get('categories') else 'CrossRef'
+        )
+
+    if form_fields.get('defense_data'):
+        builder.add_public_note('Presented on {0}'.format(
+            form_fields.get('defense_data')))
+
+    if form_fields.get('url'):
+        builder.add_url(url=form_fields.get('url'))
+
+    if form_fields.get('additional_url'):
+        builder.add_url(url=form_fields.get('additional_url'))
+
+    if form_fields.get('report_numbers'):
+        builder.add_report_numbers(report_numbers=form_fields.get('report_numbers'))
+
+    if form_fields.get('collaboration'):
+        builder.add_collaboration(collaboration=form_fields.get('collaboration'))
+
     try:
-        email = User.query.get(userid).email
+        email = User.query.get(obj.id_user).email
     except AttributeError:
-        email = ''
+        email = None
     try:
-        # TODO Make sure we are getting the ORCID id correctly
-        source = UserIdentity.query.filter_by(id_user=userid, method='orcid').one()
+        orcid = UserIdentity.query.filter_by(
+            id_user=obj.id_user,
+            method='orcid'
+        ).one().id
     except NoResultFound:
-        source = ''
-    if source:
-        source = source.method + ':' + source.id
-    data['acquisition_source'] = dict(
-        source=source,
+        orcid = None
```

Comment:
@kaplun could you suggest me the right module where I can put this util function?